src/main/java/org/folio/rest/support/HridManager.java

```diff
@@ -152,6 +154,12 @@ public HridManager(Context context, PostgresClient postgresClient) {
     return promise.future().map(v -> null);
   }
 
+  private void updateRetainLeadingZeroes(HridSettings hridSettings) {
+    hridSettings.getInstances().setRetainLeadingZeroes(hridSettings.getCommonRetainLeadingZeroes());
+    hridSettings.getHoldings().setRetainLeadingZeroes(hridSettings.getCommonRetainLeadingZeroes());
+    hridSettings.getItems().setRetainLeadingZeroes(hridSettings.getCommonRetainLeadingZeroes());
+  }
```

Comment:
This seems to overwrite the settings for each record with a single general setting. Have I understood this correctly?

Reply:
Yes. Because I added one common parameter for UI and I added individual parameters for Instances/Holdings and Items. Now we will manage "Retain Leading Zeroes" by the one checkbox for all, but in the future we can to add individual checkbox for every group. 
In addition this way contains minimal number of changes in the module HridManager, by another way the number of changes that would be required would be much more.

Reply:
Thanks for clarifying. I think this logic makes those individual parameters redundant and introduces unnecessary complexity into the code.

Given that they have been superseded by a single parameter, please remove them and update the code / upgrade scripts accordingly.

Reply:
I would prefer to leave the current implementation because all HridManager methods are designed to use separate HridSetting classes. Changing the code from using HridSetting to HridSetting**s** will require a lot of changes.
For an example, please look at the getNextHrid(final HridSetting hridSetting) method...) 
HridSetting is a main unit that provides us with the HridManager. Removing individual parameters will require serious changes in the HridManager module and tests. 
Maybe for this method I will add detailed comment for code clarifying.

Reply:
> I would prefer to leave the current implementation because all HridManager methods are designed to use separate HridSetting classes. Changing the code from using HridSetting to HridSettings will require a lot of changes.

I understand that it would be more convenient to have these properties on those classes. Unfortunately, due to the way the code generation works, that is only possible by extending the public API.

I think it is confusing and misleading to change the public API for this to include properties whose values are not respected and not how the API is intended to be used.

Reply:
@Aliaksandr-Fedasiuk Why was this comment marked as resolved? 

Is the resolution that it has been decided by the Folijet team to keep the changes to the public API?

Reply:
This is solution is ok for our team because in the future we have plans to add such a property to every type separately. The refactoring of HridManager is redundant at this time.

Reply:
> in the future we have plans to add such a property to every type separately.

If that is the case, why not only introduce the setting for each type in the back end API and have the front end set all of them from the same element in the form presented to the user?

By introducing the single setting into the back end API a change to settings per record type would require compatibility breaking changes in the future.

Reply:
This solution is the fastest and requires minimal changes, and when switching to the settings for each record, we will only need to delete one field from the database (because the system does not support the Transient functionality) and this method.

Reply:
> This solution is the fastest and requires minimal changes

By this solution, are you referring to introducing both record type specific settings *and* a general setting in the API?

> and when switching to the settings for each record, we will only need to delete one field from the database (because the system does not support the Transient functionality) and this method.

The removal of the shared setting property from the public API is a compatibility breaking change and would require synchronised changes which have caused challenges in the past. We try to avoid these where possible.

What do you mean by `transient functionality`?

Reply:
About Transient. I meant that would be more comfortable to have some property in raml-json like "@Transient" in JPA to ignore a field from persisting.

Reply:
> About Transient. I meant that would be more comfortable to have some property in raml-json like "@transient" in JPA to ignore a field from persisting.

Ok, thank you for explaining that more.

Reply:
An equivalent for a transient should be readonly=true, I think. See `effectiveCallNumberComponents` in `item.json` as an example.

Reply:
@bohdan-suprun, thank you for explanation.

Reply:
Completed. Per record-type parameters were deleted.